### PR TITLE
making change for proper formatting of kinesis stream names

### PR DESCRIPTION
**Pull request by ryandeivert**
to: @Ryxias / @chunyong-lin 
cc: @airbnb/streamalert-maintainers

## Background

Kinesis stream names have a nasty underscore in `stream_alert`.

## Changes

* Updating the default kinesis stream name to use `streamalert` instead of `stream_alert`.
* Removing the unnecessary (redundant) `_kinesis` suffix from the default stream name.
* Adding support for supplying a custom stream name, effectively overridden by the above.

## Testing

* Updating all unit tests for above changes.

### CODE REVIEW

**Comment by ryandeivert** [comment]
@chunyong-lin / @Ryxias this is ready for review - can you PTAL?

File: streamalert_cli/terraform/monitoring.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def generate_monitoring(cluster_name, cluster_dict, config):
 
     if monitoring_config.get('kinesis_alarms_enabled', True):
         cluster_dict['module']['cloudwatch_monitoring_{}'.format(cluster_name)].update({
-            'kinesis_stream': '{}_{}_stream_alert_kinesis'.format(prefix, cluster_name),
+            'kinesis_stream': '${{module.kinesis_{}.stream_name}}'.format(cluster_name),
```
Comment (chunyong-lin) [inline streamalert_cli/terraform/monitoring.py]: Should we keep `prefix` here?

Comment (ryandeivert) [reply streamalert_cli/terraform/monitoring.py]: @chunyong-lin this does effectively still use the prefix - except it uses the output from here:

https://github.com/airbnb/streamalert/pull/1027/files/9f738ef03c371190e9540fc0cdecd04c9df10fd5#diff-517d65d9de13bb32b0329474e9b11627R6

which gets its stream name from here:

https://github.com/airbnb/streamalert/pull/1027/files/9f738ef03c371190e9540fc0cdecd04c9df10fd5#diff-916958fe1956202328ac3949bb5ea39cR33

Comment (ryandeivert) [reply streamalert_cli/terraform/monitoring.py]: this change avoids defining the "stream name" multiple places.. potentially missing one if the other changes

**Pull request merged**